SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/WorkingSetSystem.cpp

```diff
@@ -0,0 +1,445 @@
+				}
+			}
+
+			const FWorkingSetData& Updated =
+				ResolvedChanges.WorkingSetUpdates.Emplace(MarkerEntityId, FWorkingSetData{ Data, Data.Request });
+
+			TSet<Worker_EntityId_Key> SetReleasedEntities;
+			TSet<Worker_EntityId_Key> SetCapturedEntities;
+			if (MaybePreviousData)
+			{
+				SetReleasedEntities = MaybePreviousData->CurrentlyApprovedRequest.RequestedEntities.Difference(
+					Updated.CurrentlyApprovedRequest.RequestedEntities);
+				SetCapturedEntities = Updated.CurrentlyApprovedRequest.RequestedEntities.Difference(
+					MaybePreviousData->CurrentlyApprovedRequest.RequestedEntities);
+			}
+			else
+			{
+				SetCapturedEntities = Updated.CurrentlyApprovedRequest.RequestedEntities;
+			}
+
+			for (const Worker_EntityId_Key CapturedEntityId : SetCapturedEntities)
+			{
+				bool bAlreadyInSet = false;
+				NewlyCapturedEntities.Emplace(CapturedEntityId, &bAlreadyInSet);
+				check(!bAlreadyInSet);
+
+				checkf(!ResolvedActorOwnershipUpdates.Contains(CapturedEntityId),
+					   TEXT("Trying to assign an entity to multiple working sets! EntityId: %lld NewMarkerEntityId: %lld "
+							"OldMarkerEntityId: %lld"),
+					   CapturedEntityId, MarkerEntityId, ResolvedActorOwnershipUpdates[CapturedEntityId]);
+				ResolvedActorOwnershipUpdates.Emplace(CapturedEntityId, MarkerEntityId);
+			}
+			for (const Worker_EntityId_Key ReleasedEntityId : SetReleasedEntities)
+			{
+				NewlyReleasedEntities.Emplace(ReleasedEntityId);
+			}
+			const Worker_EntityId* ChangedLeader = Updates.LeaderChanges.Find(MarkerEntityId);
+			if (ChangedLeader != nullptr)
+			{
+				ResolvedChanges.NewLeaders.Emplace(*ChangedLeader);
+			}
+		}
+	}
+
+	for (const auto& ReleasedEntity : Updates.RequestedActorMembershipReleases)
+	{
+		NewlyReleasedEntities.Emplace(ReleasedEntity.Key);
+	}
+
+	auto UpdatedEntities = NewlyReleasedEntities.Intersect(NewlyCapturedEntities);
+	auto CapturedEntities = NewlyCapturedEntities.Difference(NewlyReleasedEntities);
+	auto ActuallyReleasedEntities = NewlyReleasedEntities.Difference(NewlyCapturedEntities);
+
+	ResolvedChanges.ReleasedEntities = MoveTemp(ActuallyReleasedEntities);
+	ResolvedChanges.CapturedEntities = MoveTemp(CapturedEntities);
+	return ResolvedChanges;
+}
+
+void FWorkingSetSystem::UpdateMigratingEntities(const TSet<Worker_EntityId_Key>& CurrentlyMigratingEntities, ISpatialOSWorker& Connection)
+{
+	for (auto& WorkingSet : WorkingSets)
+	{
+		if (WorkingSet.Value.CurrentlyApprovedRequest.Epoch == WorkingSet.Value.BaseData.Response.Epoch)
+		{
+			// This working set is up to date.
+			continue;
+		}
+		const FWorkingSetMarkerRequest& Request = WorkingSet.Value.CurrentlyApprovedRequest;
+		const TSet<Worker_EntityId_Key> RequestedMigratingEntities = CurrentlyMigratingEntities.Intersect(Request.RequestedEntities);
+		if (RequestedMigratingEntities.Num() != 0)
+		{
+			// Some entities from this requested set are still migrating, need to wait before confirming them.
+			continue;
+		}
```

Comment:
Consider axing this

Reply:
Decided against - this sort of logic allows us to special case always approving a working set formed by a worker auth over all the entities in the set.